src/amo/pages/UserProfileEdit/index.js

```diff
@@ -198,7 +200,13 @@ export class UserProfileEditBase extends React.Component<Props, State> {
     }
 
     if (wasUpdating && !isUpdating && !errorHandler.hasError()) {
-      history.push(`/${lang}/${clientApp}/user/${newUserId}/`);
+      let newPath =
+        location.query.to || `/${lang}/${clientApp}/user/${newUserId}/`;
+      if (!newPath.startsWith('/')) {
```

Comment:
This should prevent absolute `http://` URLs but I'd like to see a test added for that.

Will this prevent protocol-less URLs? What if `to=//attackersite.com/phishing/form`? The double slash would make that a valid URL.

Reply:
Yes, absolute URLs will get converted to relative and result in a 404 after redirection, which seems valid. I will add a test for this. A protocol-less URL would get passed to `history.push`, which then generates an error and does not do the redirect, so I'll add a `try/catch` for that.

Reply:
Ah, yes, any value of `location.query.to` that will trigger an error should be handled.